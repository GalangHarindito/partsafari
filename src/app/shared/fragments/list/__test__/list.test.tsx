import { render, screen } from "@testing-library/react";
import List from "../list";
import Card from "@/shared/components/card/card";
import "@testing-library/jest-dom";

jest.mock("@/shared/components/card/card", () => {
  return function mockCard({ data }: { data: any }) {
    return <div data-testid="mock-card">{data.task}</div>;
  };
});

describe("Fragment List", () => {
  const mockData = [
    {
      id: 1,
      category: "Work",
      task: "Complete project",
      createdAt: "2023-01-01",
      completed: false,
    },
    {
      id: 2,
      category: "Personal",
      task: "Go grocery shopping",
      createdAt: "2023-01-02",
      completed: true,
    },
  ];

  test("Render Task,category,cratedAt", () => {
    render(<List datas={mockData} />);

    expect(screen.getByText("Complete project")).toBeInTheDocument();
    expect(screen.getByText("Go grocery shopping")).toBeInTheDocument();
    
    const { container } = render(<List datas={mockData} />);
    expect(container).toMatchSnapshot();
  });

  test("Render length of the data", () => {
    render(<List datas={mockData} />);

    const cards = screen.getAllByTestId("mock-card");
    expect(cards.length).toBe(mockData.length);
  });

  test("Render the correct card", () => {
    render(<List datas={mockData} />);

    mockData.forEach((data) => {
      expect(screen.getByText(data.task)).toBeInTheDocument();
    });
  });

  test('renders "Sorry, No Data Available" when datas is an empty array', () => {

    const { getByText, container } = render(<List datas={[]} />);
    expect(container).toMatchSnapshot();
    const noDataMessage = getByText('Sorry, No Data Available');
    expect(noDataMessage).toBeInTheDocument();
  });
});
