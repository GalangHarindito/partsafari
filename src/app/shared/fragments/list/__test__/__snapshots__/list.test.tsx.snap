// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fragment List Render Task,category,cratedAt 1`] = `
<div>
  <div
    class="flex flex-col gap-y-4"
  >
    <div>
      <div
        data-testid="mock-card"
      >
        Complete project
      </div>
    </div>
    <div>
      <div
        data-testid="mock-card"
      >
        Go grocery shopping
      </div>
    </div>
  </div>
</div>
`;

exports[`Fragment List renders "Sorry, No Data Available" when datas is an empty array 1`] = `
<div>
  <div
    class="flex flex-col gap-y-4"
  >
    <h3
      class=" font-bold  xl:text-2xl lg:text-xl md:text-lg"
    >
      Sorry, No Data Available
    </h3>
  </div>
</div>
`;
